Gems/Multiplayer/Code/Source/Components/NetBindComponent.cpp

```diff
@@ -167,7 +167,7 @@ namespace Multiplayer
 
     void NetBindComponent::Deactivate()
     {
-        AZ_Assert(m_needsToBeStopped == false, "Entity appears to have been improperly deleted. Use MarkForRemoval to correctly clean up a networked entity.");
+        AZ_Assert(m_needsToBeStopped == false, "Entity (%s) appears to have been improperly deleted. Use MarkForRemoval to correctly clean up a networked entity.", GetEntity()->GetName().c_str());
```

Comment:
To be safe, you should check that the string isn't null. I don't think AZ_Assert likes having null strings passed in, as I recall. "?:" operator can be used for that.

Reply:
AZStd::string c_str will always return a non-null ptr, to a null-terminated string. However, it's unlikely, but technically possible that a component is created and not assigned to an entity, so added a null check there.